myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -60,38 +60,89 @@ public ClustersResource(MyriadConfiguration cfg,
+    @Produces(MediaType.TEXT_PLAIN)
     @Consumes(MediaType.APPLICATION_JSON)
     public Response flexUp(FlexUpClusterRequest request) {
         Preconditions.checkNotNull(request,
                 "request object cannot be null or empty");
 
-        // TODO(mohit): Validation
         LOGGER.info("Received Flexup Cluster Request");
 
-
         Integer instances = request.getInstances();
         String profile = request.getProfile();
 
-        LOGGER.info("Instances: ", instances);
-        LOGGER.info("Profile: ", profile);
+        LOGGER.info("Instances: {}", instances);
+        LOGGER.info("Profile: {}", profile);
+
+        // Validate profile request
+        Response.ResponseBuilder response = Response.status(Response.Status.ACCEPTED);
+        if (!this.profileManager.exists(profile)) {
+            response.status(Response.Status.BAD_REQUEST)
+                    .entity("Profile does not exist: " + profile);
+            LOGGER.error("Provided profile does not exist " + profile);
+        } else if (!this.isValidInstanceSize(instances)) {
+            response.status(Response.Status.BAD_REQUEST)
+                    .entity("Invalid instance size: " + instances);
+            LOGGER.error("Invalid instance size request " + instances);
+        }
 
-        this.myriadOperations.flexUpCluster(instances, profile);
-        return Response.ok().build();
+        Response returnResponse = response.build();
+        if (returnResponse.getStatus() == Response.Status.ACCEPTED.getStatusCode()) {
+            this.myriadOperations.flexUpCluster(instances, profile);
+        }
+
+        return returnResponse;
     }
 
     @Timed
     @PUT
     @Path("/flexdown")
-    @Produces(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
     @Consumes(MediaType.APPLICATION_JSON)
     public Response flexDown(FlexDownClusterRequest request) {
         Preconditions.checkNotNull(request,
                 "request object cannot be null or empty");
 
-        // TODO(mohit): Make safer.
-        this.myriadOperations.flexDownCluster(request.getInstances());
-        return Response.ok().build();
+        Integer instances = request.getInstances();
+
+        LOGGER.info("Received flexdown request");
+        LOGGER.info("Instances: " + instances);
+
+        Response.ResponseBuilder response = Response.status(Response.Status.ACCEPTED);
+
+        if (!this.isValidInstanceSize(instances)) {
+            response.status(Response.Status.BAD_REQUEST)
+                    .entity("Invalid instance size: " + instances);
+            LOGGER.error("Invalid instance size request " + instances);
+        }
+
+        // warn that number of requested instances isn't available
+        // but instances will still be flexed down
+        Integer flexibleInstances = this.getFlexibleInstances();
+        if (flexibleInstances < instances)  {
+            response.entity("Number of requested instances is greater than available.");
```

Comment:
nit - will be good to return the available and requested instances in response, just like you are logging it. Also, we should return a message that says the flexdown works on a best effort basis. for e.g. if 3 instances are running and 5 are requested for flexdown, then 3 will be flexed down.